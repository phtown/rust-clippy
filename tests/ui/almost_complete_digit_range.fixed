// run-rustfix
// edition:2018
// aux-build:macro_rules.rs

#![feature(exclusive_range_pattern)]
#![feature(stmt_expr_attributes)]
#![warn(clippy::almost_complete_digit_range)]
#![allow(ellipsis_inclusive_range_patterns)]
#![allow(clippy::needless_parens_on_range_literals)]

#[macro_use]
extern crate macro_rules;

macro_rules! zero {
    () => {
        '0'
    };
}

macro_rules! b {
    () => {
        let _ = '0'..='9';
    };
}

fn main() {
    #[rustfmt::skip]
    {
        let _ = ('0') ..='9';
        let _ = '0' ..= ('9');
        let _ = ('0') ..= ('9');
    }

    let _ = '1'..'9';

    let _ = (b'0')..=(b'9');
    let _ = b'0'..=b'9';

    let _ = b'1'..b'9';

    let _ = zero!()..='9';

    let _ = match 0u8 {
        b'0'..=b'9' if true => 1,
        b'1'..b'9' => 3,
        _ => 5,
    };
    let _ = match '8' {
        '0'..='9' if true => 1,
        '1'..'9' => 3,
        _ => 5,
    };

    almost_complete_digit_range!();
    b!();
}

#[clippy::msrv = "1.25"]
fn _under_msrv() {
    let _ = match '0' {
        '0'...'9' => 1,
        _ => 2,
    };
}

#[clippy::msrv = "1.26"]
fn _meets_msrv() {
    let _ = '0'..='9';
    let _ = match '0' {
        '0'..='9' => 1,
        _ => 2,
    };
}
