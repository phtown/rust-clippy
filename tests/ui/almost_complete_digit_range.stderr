error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:29:17
   |
LL |         let _ = ('0') ..'9';
   |                 ^^^^^^--^^^
   |                       |
   |                       help: use an inclusive range: `..=`
   |
   = note: `-D clippy::almost-complete-digit-range` implied by `-D warnings`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:30:17
   |
LL |         let _ = '0' .. ('9');
   |                 ^^^^--^^^^^^
   |                     |
   |                     help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:31:17
   |
LL |         let _ = ('0') .. ('9');
   |                 ^^^^^^--^^^^^^
   |                       |
   |                       help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:36:13
   |
LL |     let _ = (b'0')..(b'9');
   |             ^^^^^^--^^^^^^
   |                   |
   |                   help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:37:13
   |
LL |     let _ = b'0'..b'9';
   |             ^^^^--^^^^
   |                 |
   |                 help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:41:13
   |
LL |     let _ = zero!()..'9';
   |             ^^^^^^^--^^^
   |                    |
   |                    help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:44:9
   |
LL |         b'0'..b'9' if true => 1,
   |         ^^^^--^^^^
   |             |
   |             help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:49:9
   |
LL |         '0'..'9' if true => 1,
   |         ^^^--^^^
   |            |
   |            help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:22:17
   |
LL |         let _ = '0'..'9';
   |                 ^^^--^^^
   |                    |
   |                    help: use an inclusive range: `..=`
...
LL |     b!();
   |     ---- in this macro invocation
   |
   = note: this error originates in the macro `b` (in Nightly builds, run with -Z macro-backtrace for more info)

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:61:9
   |
LL |         '0'..'9' => 1,
   |         ^^^--^^^
   |            |
   |            help: use an inclusive range: `...`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:68:13
   |
LL |     let _ = '0'..'9';
   |             ^^^--^^^
   |                |
   |                help: use an inclusive range: `..=`

error: almost complete ascii digit range
  --> $DIR/almost_complete_digit_range.rs:70:9
   |
LL |         '0'..'9' => 1,
   |         ^^^--^^^
   |            |
   |            help: use an inclusive range: `..=`

error: aborting due to 12 previous errors

